### PR TITLE
Fix deprecation warning and errors with libtorrent 1.1

### PR DESCRIPTION
**Pull request by rnhmjoj**
Currently btfs does not build with libtorrent 1.1.0. I've managed to fix most of the errors and warnings due to the changes in the API. There are still a couple things left to do: fix the `pop_alerts()` deprecation warning and set the min/max port when initializing the session.
Note that I haven't checked if it actually works or compile with previous versions of libtorrent.

### CODE REVIEW

File: src/btfs.cc [hunk 1]
```diff
@@ -220,7 +228,11 @@ setup() {
 		free(p);
 
 		// Path <-> file index mapping
+#if LIBTORRENT_VERSION_NUM < 10100
 		files["/" + ti.file_at(i).path] = i;
+#else
+	  files["/" + ti.files().file_path(i)] = i;
```
Comment (johang) [inline src/btfs.cc]: Wrong indentation.

File: src/btfs.cc [hunk 2]
```diff
@@ -516,19 +544,23 @@ btfs_init(struct fuse_conn_info *conn) {
...
 	se.announce_to_all_tiers = true;
 	se.download_rate_limit = params.max_download_rate * 1024;
 	se.upload_rate_limit = params.max_upload_rate * 1024;
-
 	session->set_settings(se);
+#endif
+
+#if LIBTORRENT_VERSION_NUM < 10100
 	session->add_dht_router(std::make_pair("router.bittorrent.com", 6881));
 	session->add_dht_router(std::make_pair("router.utorrent.com", 6881));
```
Comment (johang) [inline src/btfs.cc]: The add_dht_router calls are now gone for 1.1 builds? I don't like it.

Comment (rnhmjoj) [inline src/btfs.cc]: I know very little of both C++ and libtorrent: I was actually changing things to make it compile while reading the documentation, so I don't really know.

**Comment by johang** [comment]
Finally got the time to look at this. I'm merging.

**Pull request merged**